### PR TITLE
 AccessControl: Update permissions and scopes in docs

### PR DESCRIPTION
**Pull request by gamab**
Going over permissions and scope to make sure they are still in line with the code

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-enterprise/issues/1575

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/sources/enterprise/access-control/permissions.md [hunk 1]
```diff
@@ -21,55 +21,57 @@ scope
...
-permissions:delegate | The scope is only applicable for roles associated with the Access Control itself and indicates that you can delegate your permissions only, or a subset of it, by creating a new role or making an assignment. 
-reports:* | Indicates against what reports an action can be performed.
-service:access-control | Only relevant for provisioning and indicates that the action can be performed only for access control provisioning files.
-global:users:* | Indicates that action can be performed against users globally.
-users:* | Indicates that an action can be performed against users in organization level.
+Scopes | Descriptions
+--- | ---
+roles:* | Restrict an action to a set of roles. For example, `roles:*` matches any role, `roles:randomuid` matches only the role with UID `randomuid` and `roles:custom:reports:{editor,viewer}` matches both `custom:reports:editor` and `custom:reports:viewer` roles.
+permissions:delegate | The scope is only applicable for roles associated with the Access Control itself and indicates that you can delegate your permissions only, or a subset of it, by creating a new role or making an assignment.
```
Comment (gamab) [inline docs/sources/enterprise/access-control/permissions.md:73]: This one is causing us trouble to rewrite

File: docs/sources/enterprise/access-control/permissions.md [hunk 2]
```diff
@@ -21,55 +21,57 @@ scope
...
+reports:* | Restrict an action to a set of reports. For example, `reports:*` matches any report and `reports:1` matches the report with id `1`.
+service:access-control | Restrict an action to target only the access-control service. For example, you can use this in conjunction with the `provisioning:reload` or the `status:accesscontrol` actions.
+global:users:* | Restrict an action to a set of global users.
```
Comment (gamab) [inline docs/sources/enterprise/access-control/permissions.md:76]: Maybe add an example or two

File: docs/sources/enterprise/access-control/permissions.md [hunk 3]
```diff
@@ -21,55 +21,57 @@ scope
...
+users:* | Restrict an action to a set of users from an organization.
```
Comment (gamab) [inline docs/sources/enterprise/access-control/permissions.md:77]: Examples

File: docs/sources/enterprise/access-control/permissions.md [hunk 4]
```diff
@@ -21,55 +21,57 @@ scope
...
-service:access-control | Only relevant for provisioning and indicates that the action can be performed only for access control provisioning files.
-global:users:* | Indicates that action can be performed against users globally.
-users:* | Indicates that an action can be performed against users in organization level.
+Scopes | Descriptions
+--- | ---
+roles:* | Restrict an action to a set of roles. For example, `roles:*` matches any role, `roles:randomuid` matches only the role with UID `randomuid` and `roles:custom:reports:{editor,viewer}` matches both `custom:reports:editor` and `custom:reports:viewer` roles.
+permissions:delegate | The scope is only applicable for roles associated with the Access Control itself and indicates that you can delegate your permissions only, or a subset of it, by creating a new role or making an assignment.
+reports:* | Restrict an action to a set of reports. For example, `reports:*` matches any report and `reports:1` matches the report with id `1`.
+service:access-control | Restrict an action to target only the access-control service. For example, you can use this in conjunction with the `provisioning:reload` or the `status:accesscontrol` actions.
```
Comment (vtorosyan) [inline docs/sources/enterprise/access-control/permissions.md:75]:
```suggestion
service:accesscontrol | Restrict an action to target only the fine-grained access control service. For example, you can use this in conjunction with the `provisioning:reload` or the `status:accesscontrol` actions.
```

The `access-control` has been changed to `accesscontrol` for consistency.

File: docs/sources/enterprise/access-control/permissions.md [hunk 5]
```diff
@@ -21,55 +21,57 @@ scope
...
---- | --- 
-roles:* | Indicates against what roles an action can be performed. For example, `roles:*` assumes any roles, and `roles:randomuid` assumes only a role with UID `randomuid`.  
-permissions:delegate | The scope is only applicable for roles associated with the Access Control itself and indicates that you can delegate your permissions only, or a subset of it, by creating a new role or making an assignment. 
-reports:* | Indicates against what reports an action can be performed.
-service:access-control | Only relevant for provisioning and indicates that the action can be performed only for access control provisioning files.
-global:users:* | Indicates that action can be performed against users globally.
-users:* | Indicates that an action can be performed against users in organization level.
+Scopes | Descriptions
+--- | ---
+roles:* | Restrict an action to a set of roles. For example, `roles:*` matches any role, `roles:randomuid` matches only the role with UID `randomuid` and `roles:custom:reports:{editor,viewer}` matches both `custom:reports:editor` and `custom:reports:viewer` roles.
```
Comment (vtorosyan) [inline docs/sources/enterprise/access-control/permissions.md:72]: I really like the examples in the scopes and the new descriptions 👍

**Review by osg-grafana — APPROVED** [review]

**Review by vtorosyan — APPROVED** [review]
The access-control needs to be changed, but other then that 👍

File: docs/sources/enterprise/access-control/permissions.md (hunk 5 above)
Comment (sakjur) [inline docs/sources/enterprise/access-control/permissions.md:72]: Those are not valid UIDs post https://github.com/grafana/grafana-enterprise/issues/1466

File: docs/sources/enterprise/access-control/permissions.md [hunk 6]
```diff
@@ -21,55 +21,57 @@ scope
...
-ldap.user:sync | n/a | Allows to sync LDAP users.
-ldap.status:read | n/a | Allows to check LDAP status.
+roles:list | roles:* | List available roles without permissions.
+roles:read | roles:* | Read a specific role with it's permissions.
+roles:write | permissions:delegate | Create or update a custom role.
+roles:delete | permissions:delegate | Delete a custom role.
+roles.builtin:list | roles:* | List built-in role assignments.
+roles.builtin:add | permissions:delegate | Create a built-in role assignment.
+roles.builtin:remove | permissions:delegate | Delete a built-in role assignment.
+reports.admin:create | reports:* | Create reports.
```
Comment (gamab) [inline docs/sources/enterprise/access-control/permissions.md:35]: Need to add this the concerned HTTP endpoint

Comment (gamab) [reply docs/sources/enterprise/access-control/permissions.md:35]: I cannot find the `POST /api/reports/new` So I assume we can skip that

File: docs/sources/enterprise/access-control/permissions.md [hunk 7]
```diff
@@ -21,55 +21,57 @@ scope
...
+users.quotas:list | global:users:* | List a user’s quotas.
+users.quotas:update | global:users:* | Update a user’s quotas.
+org.users.read | users:* | Get user profiles within an organization.
+org.users.add | users:* | Add a user to an organization.
+org.users.remove | users:* | Remove a user from an organization.
+org.users.role:update | users:* | Update the organization role (`Viewer`, `Editor`, `Admin`) for an organization.
+ldap.user:read | n/a | Get a user via LDAP.
+ldap.user:sync | n/a | Sync a user via LDAP.
+ldap.status:read | n/a | Verify the LDAP servers’ availability.
+status:accesscontrol | service:access-control | Get access-control enabled status.
```
Comment (gamab) [inline docs/sources/enterprise/access-control/permissions.md:64]: Added this one. Need to double check we are ok to communicate about it. @vtorosyan ?

Comment (gamab) [reply docs/sources/enterprise/access-control/permissions.md:64]: We did advertise the enpoint in docs/source/http_api/access_control.md so I assume this is ok :)

Comment (vtorosyan) [reply docs/sources/enterprise/access-control/permissions.md:64]: yes this is needed

File: docs/sources/enterprise/access-control/permissions.md (hunk 5 above)
Comment (gamab) [reply docs/sources/enterprise/access-control/permissions.md:72]: Good point. We did not backport that change. But if we still need to change this, then conf/access-control/sample.yaml and docs/sources/enterprise/access-control/provisioning.md need to be updated

File: docs/sources/enterprise/access-control/permissions.md (hunk 6 above)
Comment (vtorosyan) [reply docs/sources/enterprise/access-control/permissions.md:35]: This endpoint is not documented in the HTTP API in the reporting page it just mentioned you need to have the right permission. For now having this mentioned in list of actions is sufficient

File: docs/sources/enterprise/access-control/permissions.md (hunk 5 above)
Comment (donomii) [reply docs/sources/enterprise/access-control/permissions.md:72]: the {editor,viewer} ?

Comment (vtorosyan) [inline docs/sources/enterprise/access-control/permissions.md:72]:
```suggestion
roles:* | Restrict an action to a set of roles. For example, `roles:*` matches any role, `roles:5vV9FheMk` matches only the role with UID `5vV9FheMk` and `roles:{PS49Fh6Gk,KHVrKheGz}` matches both roles with UID `PS49Fh6Gk` and `KHVrKheGz` .
```

File: docs/sources/enterprise/access-control/permissions.md (hunk 1 above)
Comment (donomii) [inline docs/sources/enterprise/access-control/permissions.md:73]:
```suggestion
permissions:delegate | The scope is only applicable for roles associated with the Access Control itself and indicates that you can delegate your own permissions by creating a new role or making an assignment.
```

File: docs/sources/enterprise/access-control/permissions.md (hunk 5 above)
Comment (vtorosyan) [reply docs/sources/enterprise/access-control/permissions.md:72]: I'd recommend to keep it simple for now and give an example of restricting to one role UID.

File: docs/sources/enterprise/access-control/permissions.md (hunk 1 above)
Comment (vtorosyan) [inline docs/sources/enterprise/access-control/permissions.md:73]:
```suggestion
permissions:delegate | The scope is only applicable for the fine-grained access control itself. It indicates that you create a new role or a built-in role assignments with a subset of permissions that you have.
```

**Comment by vtorosyan** [comment]
I am going to merge this when the build runs. Whatever we have is an improvement on top of what was already there.

**Comment by osg-grafana** [comment]
Thank you for the teamwork on this one, @gamab and @donomii. Made this a lot easier today. Enjoy your weekends. :)

File: docs/sources/enterprise/access-control/permissions.md (hunk 1 above)
Comment (donomii) [reply docs/sources/enterprise/access-control/permissions.md:73]:
```suggestion
permissions:delegate | This scope is only applicable for fine-grained access control. It allows you to create a new role or built-in role assignment with a subset of your own permissions.
```

Comment (donomii) [reply docs/sources/enterprise/access-control/permissions.md:73]: Tweaked vardan's a bit

**Comment by sakjur** [comment]
@vtorosyan You can press the "auto-merge" button to get that effect 🙂

File: docs/sources/enterprise/access-control/permissions.md (hunk 5 above)
Comment (sakjur) [reply docs/sources/enterprise/access-control/permissions.md:72]: Colons are not valid characters in UIDs post URI restrictions `:`. Good point about provisioning docs and samples, we'll have to fix that.

**Pull request merged**